### PR TITLE
Store: Add promotions selector

### PR DESCRIPTION
**Pull request by coderkevin**
This adds a selector for the promotions list that is compiled via
API data

To Test: `npm test`

### CODE REVIEW

**Comment by coderkevin** [comment]
Rebased.

File: client/extensions/woocommerce/state/selectors/promotions.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteWithFallback } from '../sites/selectors';
+
+export function getPromotions( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const { promotions } = rootState.extensions.woocommerce.sites[ siteId ];
+	return promotions.promotions;
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/selectors/promotions.js]: Should we return an empty array here if no promotions are found? We do this for our other similar selectors.

Comment (allendav) [reply client/extensions/woocommerce/state/selectors/promotions.js]: FWIW, use have used `lodash` `get` for things like this (both getting the item from state and returning an appropriate default)

File: client/extensions/woocommerce/state/selectors/test/promotions.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+} from '../promotions';
+
+describe( 'promotions', () => {
+	const rootState = {
+		extensions: {
+			woocommerce: {
+				sites: {
+					123: {
+						promotions: {
+							promotions: [
```
Comment (allendav) [inline client/extensions/woocommerce/state/selectors/test/promotions.js]: Just curious - why doubly nested `promotions` keys?

Comment (allendav) [reply client/extensions/woocommerce/state/selectors/test/promotions.js]: Nevermind. https://github.com/Automattic/wp-calypso/pull/18295/files answered my question

File: client/extensions/woocommerce/state/selectors/promotions.js (hunk 1 above)
Comment (coderkevin) [reply client/extensions/woocommerce/state/selectors/promotions.js]: In this case, I want the `undefined` return when the promotions haven't yet been loaded, so I can reflect that with placeholders in the UI.

Comment (coderkevin) [reply client/extensions/woocommerce/state/selectors/promotions.js]: > FWIW, use have used lodash get...

Yeah, I don't always bring it in until I have more need for it. But I think I add it in a later PR anyway...

**Comment by coderkevin** [comment]
Rebased.

**Pull request merged**